include/oneapi/dpl/pstl/hetero/dpcpp/sycl_traits.h

```diff
@@ -221,6 +224,12 @@ struct sycl::is_device_copyable<_ONEDPL_SPECIALIZE_FOR(oneapi::dpl::__internal::
 {
 };
 
+template <typename... _Types>
+struct sycl::is_device_copyable<_ONEDPL_SPECIALIZE_FOR(oneapi::dpl::__internal::tuple, _Types...)>
```

Comment:
Sorry, but this fix seems very weird to me. Why do we need this specialization at all, if `internal::tuple` is trivially copyable if all the fields are trivially_copyable?

Reply:
This is how we missed it the first time around.  Consider the following example
`internal::tuple<std::tuple<int>>`

Is it trivially copyable?  it depends on `std::tuple` implementation, implementations we use often, no.  Should it be device copyable? yes.

This is why we need the specialization, for non-trivially copyable but still device copyable types.

Reply:
`std::tuple` is not unique here also, all other types in this file may also be in the same situation, if some type they depend upon is not trivially copyable.

Reply:
ok, got it, thanks. I would prefer to `enable_if` this specialization only in case `_Types` are not trivially copyable, but unfortunately it's impossible to the best of our knowledge (prior C++20)

Reply:
Ah ok.  I'm not sure what exactly that buys us.  The result effectively is the same, but just a different compile time path to the same `std::true_type` result.  

I too wish this specialization was not required, most of the types we specialize in this file are trivially copyable with trivially copyable template params.